docs/develop/cron-jobs.md

```diff
@@ -0,0 +1,107 @@
+---
+title: Cron Jobs
+---
+
+You can use a `CronJob` to run jobs on a time-based schedule. These automated jobs run similarly to [Cron](https://en.wikipedia.org/wiki/Cron) tasks.
+
+Cron jobs are useful for creating periodic and recurring tasks, like running a database query or sending emails. Cron jobs can also schedule individual tasks for a specific time, such as if you want to schedule a job for a low activity period.
+
+<Callout type="important">
+
+**Cron jobs** have limitations and idiosyncrasies:
+- In certain circumstances, a single cron job can create multiple jobs. Therefore, jobs should be **idempotent**.
+- The actual job needs to be expressed as a **public** API endpoint, that doesn't have any request data.
+- You can uniquely identify a Cron job resource using its ID; this makes it easier to rename or move infrastructure around without recreating or destroying it.
+
+</Callout>
+
+## Defining a Cron job
+
+With Encore you define cron jobs directly in your code, using our infrastructure declaration syntax.
+
+```go
+package hello // service name
+
+import (
+	"context"
+	"encore.dev/cron"
+)
+
+/*
+  In this example, we are defining a cron job.
+  The comment above the cron job definition will 
+  be used as a description.
+
+  You can also add code snippets in your description,
+  which will be picked up by our parser and displayed
+  in our web UI.
+
+    var cfg := cron.JobConfig{
+      Name:     "Cron Job Example",
+      Every:    5 * time.Minute,
+      Endpoint: Cron
+    }
+*/
```

Comment:
The comment is a bit meta; it talks about itself and the code snippet example is an example of itself. Let's turn this whole example into a real-world example. https://jvns.ca/blog/2021/07/08/writing-great-examples/ has some great ideas on this topic :)